version-catalog/README.md

```diff
@@ -0,0 +1,176 @@
+their resulting accessors.
+
+Source code of `Dummy` dependency:
+
+```kotlin
+internal object Dummy : DependencyEntry() {
+
+    private const val group = "org.dummy.company"
+    override val module = "$group:dummy-lib" // libs.dummy
+    override val version = "1.0.0"           // libs.versions.dummy
+
+    val core by lib("$group:dummy-core")     // libs.dummy.core
+    val runner by lib("$group:dummy-runner") // libs.dummy.runner
+    val api by lib("$group:dummy-api")       // libs.dummy.api
+
+    // In bundles, you can reference already declared libs,
+    // or create them in-place.
+
+    override val bundle = setOf( // libs.bundles.dummy
+        core, runner, api,
+        lib("params", "$group:dummy-params"), // libs.dummy.params
+        lib("types", "$group:dummy-types"),   // libs.dummy.types
+    )
+
+    // "GradlePlugin" - is a special entry name for `PluginEntry`.
+    // For plugin entries with this name, the facade will not put "gradlePlugin"
+    // suffix for a plugin's id. Note, that we have this suffix for the version
+    // and module, and does not have for id.
+
+    object GradlePlugin : PluginEntry() {
+        override val version = "0.0.8"                 // libs.versions.dummy.gradlePlugin
+        override val module = "$group:my-dummy-plugin" // libs.dummy.gradlePlugin
+        override val id = "my-dummy-plugin"            // libs.plugins.dummy
+    }
+
+    object Runtime : DependencyEntry() {
+
+        // When an entry does not override the version, it is taken from
+        // the outer entry. For example, in this case, all libs within "Runtime"
+        // entry will have "1.0.0".
+
+        val win by lib("$group:runtime-win")     // libs.dummy.runtime.win
+        val mac by lib("$group:runtime-mac")     // libs.dummy.runtime.mac
+        val linux by lib("$group:runtime-linux") // libs.dummy.runtime.linux
+
+        object BOM : LibraryEntry() {
+            override val version = "2.0.0"           // libs.versions.dummy.runtime.bom
+            override val module = "$group:dummy-bom" // libs.dummy.runtime.bom
+        }
+    }
+
+    // A library that is declared as `object SomeLib : LibraryEntry()` can be
+    // referenced as well as the one declared by `lib()` delegate.
+
+    val runtime by bundle( // libs.bundles.dummy.runtime
+        Runtime.BOM,
+        Runtime.win,
+        Runtime.mac,
+        Runtime.linux,
+    )
+
+    // It is also possible to declare just a bare version.
+
+    object Tools : VersionEntry() {
+        override val version = "3.0.0" // libs.versions.dummy.tools
+    }
+}
+```
+
+
+## Modules structure
+
+Within this PR, `spine-version-catalog` is a resident of `time` repository, but
+it is a standalone project. Meaning, it has its own Gradle's `settings` file,
+and doesn't relate anyhow to `time`.
+
+`spine-version-catalog` consists of several modules:
+
+1. `api` – represents a facade upon Gradle's provided `VersionCatalogBuilder`. 
```

Comment:
We have discussed this previously. I would not have this module as a standalone entity.